turbopack/crates/turbopack-node/src/evaluate.rs

```diff
@@ -382,59 +354,51 @@ pub trait EvaluateContext {
     ) -> impl Future<Output = Result<()>> + Send;
 }
 
-pub async fn custom_evaluate(
-    evaluate_context: impl EvaluateContext,
-) -> Result<Vc<JavaScriptEvaluation>> {
-    // TODO: The way we invoke compute_evaluate_stream as side effect is not
-    // GC-safe, so we disable GC for this task.
-    prevent_gc();
+pub async fn custom_evaluate(evaluate_context: impl EvaluateContext) -> Result<Vc<Option<RcStr>>> {
+    let pool_op = evaluate_context.pool();
+    let mut state = Default::default();
+
+    // Read this strongly consistent, since we don't want to run inconsistent
+    // node.js code.
+    let pool = pool_op.read_strongly_consistent().await?;
+
+    let args = evaluate_context.args().iter().try_join().await?;
+    // Assume this is a one-off operation, so we can kill the process
+    // TODO use a better way to decide that.
+    let kill = !evaluate_context.keep_alive();
+
+    // Workers in the pool could be in a bad state that we didn't detect yet.
+    // The bad state might even be unnoticeable until we actually send the job to the
+    // worker. So we retry picking workers from the pools until we succeed
+    // sending the job.
+
+    let (mut operation, _) = FutureRetry::new(
+        || async {
+            let mut operation = pool.operation().await?;
+            operation
+                .send(EvalJavaScriptOutgoingMessage::Evaluate {
+                    args: args.iter().map(|v| &**v).collect(),
+                })
+                .await?;
+            Ok(operation)
+        },
+        PoolErrorHandler,
+    )
+    .await
+    .map_err(|(e, _)| e)?;
 
-    // Note the following code uses some hacks to create a child task that produces
-    // a stream that is returned by this task.
+    // The evaluation sent an initial intermediate value without completing. We'll
+    // need to spawn a new thread to continually pull data out of the process,
+    // and ferry that along.
+    let result = pull_operation(&mut operation, &pool, &evaluate_context, &mut state).await?;
```

Comment:
I only see a single `pull_operation` now. Don't we still need to call that potentially more than once?